app/components/accountForm/AccountForm.js

```diff
@@ -0,0 +1,147 @@
+import React, { PropTypes } from 'react';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+import { View, TouchableWithoutFeedback } from 'react-native';
+import { MenuContext } from 'react-native-popup-menu';
+import DateTimePicker from 'react-native-modal-datetime-picker';
+import {
+  IconsPickerModal,
+  TextFieldWithIcon,
+  Button,
+  FixedButtonsContainer,
+  SceneContentWrapper,
+} from '../';
+import CurrencyPicker from '../CurrencyPicker';
+import calendarDateFormat from '../../configs/calendarDateFormat';
+
+const AccountForm = (props) => {
+  const {
+    style: { blockStyleDark, blockStyle, rowStyle },
+    name,
+    icon,
+    date,
+    currency,
+    onSubmit,
+    isValid,
+    initialBalance,
+    onNameChange,
+    onIconChange,
+    onDateChange,
+    isPickerVisible,
+    onCurrencyChange,
+    onInitialBalanceChange,
+    onTogglePicker,
+    onToggleDatePicker,
+    isDatePickerVisible,
+  } = props;
+
+  const iconsPickerButton = (
+    <Button
+      onPress={onTogglePicker}
+      iconsSet={MaterialCommunityIcons}
+      icon={icon}
+      raised
+    />
+  );
+
+  const pickerIcon = (
+    <Button
+      icon={'calendar-blank'}
+      iconsSet={MaterialCommunityIcons}
+      onPress={onToggleDatePicker}
+    />
+  );
+
+  const submitButton = (
+    <FixedButtonsContainer>
+      <Button
+        onPress={() => onSubmit(props)}
```

Comment:
onPress={onSubmit} and just refactor for this onSubmit method in accountForm/index.js